mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/mapping/definition/annotation/VectorField.java

```diff
@@ -93,6 +93,14 @@
 	 */
 	int m() default AnnotationDefaultValues.DEFAULT_M;
 
+	/**
+	 * @return The vector magnitude.
+	 * <p>
+	 * With float vectors, if set to {@code > 0} automatic normalization will be applied, i.e. {@code ||vector|| == magnitude }.
+	 * With byte vectors, if set to {@code > 0} a check will be performed that the passed vector is of required magnitude.
+	 */
+	float magnitude() default AnnotationDefaultValues.DEFAULT_MAGNITUDE;
```

Comment:
I think it's fine to require normalization to be explicit, but you might want to make the checks implicit?
Maybe `@VectorField(magnitude = ..., normalization = ...)` with normalization defined as `enum Normalization { DEFAULT, CHECK, NORMALIZE }`, and `DEFAULT` effectively identical to `CHECK` for now, at least for the Lucene backend?

Reply:
> `DEFAULT` effectively identical to `CHECK` for now, at least for the Lucene backend?

Scratch that, that's dumb. Better idea would be to have `DEFAULT` be equal to `CHECK` for similarities that require normalization, and `IGNORE` otherwise.